### PR TITLE
Rename net.minecraft.world.gen.decorator package to placementmodifier

### LINKED ISSUE #2862: The package `net.minecraft.world.gen.decorator` should be renamed to placement modifier
It's called placement modifier now.

### PR DESCRIPTION
**Pull request by NoahvdAa**
Resolves #2862.

### CODE REVIEW

File: mappings/net/minecraft/world/gen/placementmodifier/RangeDecoratorConfig.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_2997 net/minecraft/world/gen/decorator/RangeDecoratorConfig
+CLASS net/minecraft/class_2997 net/minecraft/world/gen/placementmodifier/RangeDecoratorConfig
```
Comment (enbrain) [inline mappings/net/minecraft/world/gen/placementmodifier/RangeDecoratorConfig.mapping]: This is actually a `FeatureConfig` (a config used to make `ConfiguredFeature` from `Feature`). Should probably be `net.minecraft.world.gen.RangeFeatureConfig`.

**Review by enbrain — COMMENTED** [review]
Some of the placement modifier classes (such as `AbstractCountPlacementModifier`) are still in the decorator package.

**Comment by NoahvdAa** [comment]
Thanks, don't know how I missed that.

**Review by enbrain — COMMENTED** [review]
`DecoratorContext` should be renamed too. It's a context passed to `PlacementModifier#getPositions`. Maybe `net.minecraft.world.gen.placementmodifier.PlacementModifierContext`? (Is this a little bit too long?)

**Comment by Juuxel** [comment]
Or `FeaturePlacementContext` could work too (a plain `PlacementContext` won't work since it can be confused with `ItemPlacementContext`)

**Comment by NoahvdAa** [comment]
> Or `FeaturePlacementContext` could work too (a plain `PlacementContext` won't work since it can be confused with `ItemPlacementContext`)

:+1: Changed to FeaturePlacementContext since it's not only used in placement modifiers.

**Review by enbrain — APPROVED** [review]

File: mappings/net/minecraft/world/gen/RangeFeatureConfig.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_2997 net/minecraft/world/gen/decorator/RangeDecoratorConfig
+CLASS net/minecraft/class_2997 net/minecraft/world/gen/RangeFeatureConfig
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/RangeFeatureConfig.mapping]: This should be in the feature package

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by Juuxel — APPROVED** [review]

**Review by liach — APPROVED** [review]
Good by naked eye

**Review by Shnupbups — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**